turbinia/workers/binary_extractor.py

```diff
@@ -41,21 +43,44 @@ def run(self, evidence, result):
     binary_extraction_evidence = BinaryExtraction()
 
     binary_extraction_evidence.local_path = self.output_dir
-    image_export_log = os.path.join(
-        self.output_dir, '{0:s}.log'.format(self.id))
+    image_export_log = os.path.join(self.output_dir, 'binary_extraction.log')
+
+    binary_extraction_dir = os.path.join(self.output_dir, 'extracted_binaries')
 
     cmd = [
-        'image_export.py', '--signatures', 'elf,exe_mz', '--logfile',
-        image_export_log
+        'image_export.py', '--partitions', 'all', '--no_vss', '--signatures',
+        'elf,exe_mz', '--logfile', image_export_log
     ]
     if config.DEBUG_TASKS:
       cmd.append('-d')
-    cmd.extend(['-w', self.output_dir, evidence.local_path])
+    cmd.extend(['-w', binary_extraction_dir, evidence.local_path])
 
     result.log('Running image_export as [{0:s}]'.format(' '.join(cmd)))
 
     self.execute(
         cmd, result, log_files=[image_export_log],
```

Comment:
I would consider adding your JSON log file to this as well, and that way it will get saved separately, and you won't need to extract the archive to see it.

Reply:
JSON report is always saved in binary_extraction_dir folder, image_export.py does not have a flag to save it separately. We might add it though to image_export.py at some point. Currently, turbinia_output contain following structure for this job:

├── 1572523833-32098d068ce84cb690534c89460ad0bb-BinaryExtractorTask
│   ├── binary_extraction.log
│   ├── extracted_binaries
│   │   ├── bin
│   │   │   ├── bzcat
│   │   │   └── echo
│   │   ├── hashes.json
│   │   ├── home
│   │   │   └── echo
│   │   └── sbin
│   │       ├── tune2fs
│   │       └── visudo
│   └── worker-log.txt
├── 1572523833-32098d068ce84cb690534c89460ad0bb-BinaryExtractorTask.tar.gz

If my understanding is correct, we can still get hashes.json report without looking into .tar.gz file.

Reply:
That's nice, I've added json_path to log_files and moved json_path and binary_extraction_dir variables to instance level, so I don't have to repeat 

json_path = os.path.join(self.binary_extraction_dir, 'hashes.json')

in both run() and check_extraction()